projects/core/src/product/facade/product-search.service.ts

```diff
@@ -31,6 +34,15 @@ export class ProductSearchService {
   }
 
   getResults(): Observable<ProductSearchPage> {
+    const results = this.getResultsFromTransferState();
+
+    if (results) {
+      return merge(
+        this.store.pipe(select(ProductSelectors.getSearchResults)),
```

Comment:
I wonder won't this trigger a request?

Reply:
I don't think that selectors do like actions.

Reply:
Right, my mistake. How do you stop the request then, I believe the `search` function is triggered in ngOnInit in one of the components. I may be missing something here.

Reply:
In short, in the product list component, it will call the `search()` function in this product search service only if the results are not populated in the store. The results are got with this `getResults()` method. 

This fix will return the transfer state products if they exist along with the products in the store. So search won't get called again if it is already done on the server-side.